raw/gcs_data_consistency_checker.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import fsspec
+import pandas as pd
+import typing as t
+
+BUCKET = "gcp-public-data-arco-era5"
+START_YEAR = 1979
+END_YEAR = 2022
+
+
+def list_files_in_directory(bucket_name: str, directory_prefix: str) -> t.List[str]:
+    """List files in a directory of a cloud storage bucket using fsspec.
+
+    Args:
+        bucket_name (str): The name of the cloud storage bucket.
+        directory_prefix (str): The prefix of the directory.
+
+    Returns:
+        List[str]: A list of file names in the directory.
+    """
+    fs = fsspec.filesystem('gs')
+    files = fs.ls(f'{bucket_name}/{directory_prefix}')
+    return [file[len(bucket_name) + 1:] for file in files]
+
+
+def find_extra_files(directory_files: t.List[str], desired_files: t.List[str]
+                     ) -> t.List[str]:
+    """Returns a list of files that are in the directory but not in the desired list.
+
+    Args:
+        directory_files (List[str]): List of files in the directory.
+        desired_files (List[str]): List of desired file names.
+
+    Returns:
+        List[str]: A list of file names that are extra in the directory.
+    """
+    return list(set(directory_files) - set(desired_files))
+
+
+def generate_daily_file_names(year: int, chunks: t.List) -> t.List[str]:
+    """Generate a list of daily file names for a given year.
+
+    Args:
+        year (int): The year for which to generate file names.
+        chunks (List):  A list of the daily files chunks.
+    Returns:
+        List[str]: A list of generated daily file names.
+    """
+    generated_files = []
+    start_date = pd.Timestamp(year, 1, 1)
+    end_date = pd.Timestamp(year + 1, 1, 1)
+    date_range = pd.date_range(start_date, end_date, freq='D')
```

Comment:
Optional: `date_range()` can take ISO date strings. So, this could be: 

```
date_range = pd.date_range(f'{year}-01-01', f'{year+1}-01-01', freq='D')
```
or something similar. I think this could save a few LOC :)